[FIX] web_editor: fix reset when triggering scroll

Iucapad opened the pull request:
This commit fixes the behavior hiding the select element when scrolling has happened. Instead of the 'scroll' event, 'mousewheel' was used, which isn't triggered when scrolling on a touch device.

Now, the correct event is used, and the listener is given to the actual document element, instead of being globally listening on window (in case of iframes).

This fix is part of a bug happening in the report editor of Studio, since the select element wouldn't be reset when scrolling, but kept visible at the same outdated position.

ticket-3496665

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/143313).

Iucapad commented:
Hello @Goaman. May I ask your review for this fix?

I changed "mousewheel" to "scroll", and add the event listener to the current document. I also moved the addEventListener in the code but in the end, this make the select element hidden once a user scrolls as expected

Goaman commented:
@Iucapad lgtm 👍

aab-odoo reviewed (approved):
robodoo r+
